src/matcher/index.ts

```diff
@@ -462,4 +485,20 @@ function checkMissingParamsInAbsolutePath(
   }
 }
 
+/**
+ * Allows differentiating lazy components from functional components and vue-class-component
+ *
+ * @param component
+ */
+function isRouteComponent(
```

Comment:
Import it instead of redeclaring it

Reply:
It is slightly different `isRouteComponent` declared in `navigationGuard`. 😃

```javascript
Object.keys(component).length === 0 // in matcher/index.ts
typeof component === 'object'  // in navigationGuards.ts 
```

Reply:
It may cause confutsion i changed name of function. and slightly change logic of guard.

Reply:
isRawRouteComponent is added. 

```javascript
function isRawRouteComponent(
  component: RawRouteComponent
): component is RouteComponent {
  return (
    typeof component === 'object' ||
    typeof component === 'function' || //  ⭐  This was added because of () => import('./Home.vue')
    'displayName' in component ||
    'props' in component ||
    '__vccOpts' in component
  )
}
```

Reply:
It can't be the same as  `isRouteComponent declared in navigationGaurds`.

or i can change `isRouteComponent declared in navigationGuards` same as above